File: strings/string_case/case.rb
class String
  def is_upper?
    self == self.upcase
  end

  def is_lower?
    self == self.downcase
  end

  ## returns <MatchData "some string">
  # !#<MatchData "jz"> -> false
  # !!#<MatchData "jz"> -> true
  # match returns matchdata or nil
  # !nil vs !!nil
  # https://ruby-doc.org/core-2.1.1/Regexp.html#class-Regexp-label-Character+Properties

  def is_alphanum?
    !!match(/^[[:alnum:]]+$/)
  end

  def is_alpha?
    !!match(/^[[:alpha:]]+$/)
  end
end

File: strings/string_case/case.py
import sys
for line in sys.stdin:
    args = [line]
string = args[0]

p = 'False'
for s in range(len(string)):
    if string[s-1].isalnum():
        p = 'True'
print(p)

p = 'False'
for s in range(len(string)):
    if string[s-1].isalpha():
        p = 'True'
print(p)

p = 'False'
for s in range(len(string)):
    if string[s-1].isdigit():
        p = 'True'
print(p)

p = 'False'
for s in range(len(string)):
    if string[s-1].islower():
        p = 'True'
print(p)

p = 'False'    
for s in range(len(string)):
    if string[s-1].isupper():
        p = 'True'
print(p)
